src/Cake.Issues.Reporting.Sarif/SarifIssueReportGenerator.cs

```diff
@@ -47,18 +47,46 @@ protected override FilePath InternalCreateReport(IEnumerable<IIssue> issues)
     {
         this.Log.Information("Creating report '{0}'", this.Settings.OutputFilePath.FullPath);
 
+        IEnumerable<SarifIssue> sarifIssues;
+
+        if (this.sarifIssueReportFormatSettings.ExistingIssues.Count > 0)
+        {
+            var issueComparer = new IIssueComparer();
```

Comment:
The question is if changes to e.g. location inside a file is what counts as `significant` as described in the specification

Reply:
> The question is if changes to e.g. location inside a file is what counts as `significant` as described in the specification: https://docs.oasis-open.org/sarif/sarif/v2.0/csprd02/sarif-v2.0-csprd02.html#_Toc10127853

The specification says that it's hard to find an algorithm to find out whats a significant change.
There is also written

*A result management system SHOULD NOT include an absolute line number (or an absolute byte location in a binary artifact) in its fingerprint computation.*

So the issue comparer supports this with the flag `compareOnlyPersistentProperties`.
The issue comparer also ignores the `ProjectFileRelativePath` and `AffectedFileRelativePath` so the question is, should we also use this two properties for the sarif report or not. Imho we should not make a difference between sarif and other reports. That means we using the comparer as it is, or remove this two properties and only ignore the line number...

What's your suggestion?

Reply:
Considering an issue to be updated if file path or line number changes sounds good to me. 

Fingerprint is not that easy. Hashcode of `IIssueComparer` is not stable across different versions (e.g. if additional properties are added to `IIssue`). It can be versioned (like `stableResultHash/v2`), similar to the different serialization formats, but a version of Cake.Issue won't be able to generate hash code of previous major versions (e.g. if a proprty is removed from `IIssue`). But this is something to discuss in #639. If there is a fingerprint implementation in the future the code to determine `BaselineState` can be updated to make use of fingerprints. But as long as #639 is not implemented I would stick with the logic you've implemented.